### PR TITLE
Tbb 2016

### PR DESCRIPTION
**Pull request by jas02**
Review please.

### CODE REVIEW

File: easybuild/easyblocks/t/tbb.py [hunk 1]
```diff
@@ -48,18 +51,38 @@ def __init__(self, *args, **kwargs):
...
             }
 
-        super(EB_tbb, self).install_step(silent_cfg_names_map=silent_cfg_names_map)
+        if LooseVersion(self.version) < LooseVersion('4.4'):
+            silent_cfg_names_map = {
+                'install_mode_name': INSTALL_MODE_NAME_2015,
+                'install_mode': INSTALL_MODE_2015,
+            }
+
+        # In case of TBB 2016 we have to specify ARCH_SELECTED in silent.cfg
```
Comment (boegel) [inline easybuild/easyblocks/t/tbb.py]: the version is actually 4.4.x, so change this comment to

```
# for TBB 4.4.x and newer we have to specify ARCH_SELECTED in silent.cfg
```

File: easybuild/easyblocks/t/tbb.py [hunk 2]
```diff
@@ -48,18 +51,38 @@ def __init__(self, *args, **kwargs):
         """Initialisation of custom class variables for tbb"""
         super(EB_tbb, self).__init__(*args, **kwargs)
         self.libpath = 'UNKNOWN'
+        platform_name = get_platform_name()
+        if platform_name.startswith('x86_64'):
+            self.arch = "intel64"
+        elif platform_name.startswith('i386'):
```
Comment (boegel) [inline easybuild/easyblocks/t/tbb.py]: please make this

```python
elif platform_name.startswith('i386') or platform_name.startswith('i686'):
```

and make the same change in ipp.py please

**Comment by boegel** [comment]
Jenkins: ok to test

**Comment by boegel** [comment]
jas02: some minor remarks, looks good

File: easybuild/easyblocks/t/tbb.py (hunk 2 above)
Comment (jas02) [inline easybuild/easyblocks/t/tbb.py]: Done, same for ipp.py

File: easybuild/easyblocks/t/tbb.py [hunk 3]
```diff
@@ -48,18 +51,38 @@ def __init__(self, *args, **kwargs):
...
+        if platform_name.startswith('x86_64'):
+            self.arch = "intel64"
+        elif platform_name.startswith('i386') or platform_name.startswith('i686'):
+            self.arch = 'ia32'
+        else:
+            raise EasyBuildError("Failed to determine system architecture based on %s", platform_name)
 
     def install_step(self):
         """Custom install step, to add extra symlinks"""
         silent_cfg_names_map = None
```
Comment (boegel) [inline easybuild/easyblocks/t/tbb.py]: please add `silent_cfg_extras = None` here too

Comment (jas02) [inline easybuild/easyblocks/t/tbb.py]: Done, sorry for bug.

**Comment by boegel** [comment]
merged via #769, thanks @jas02

**Pull request merged**